diff --git a/content/public/browser/tracing_support.cc b/content/public/browser/tracing_support.cc
index 66feb5091cc9436cbb12a83434ac748c1549149f..2aae5c1fb4a596c36e934092190dd6571ec9f5c4 100644
--- a/content/public/browser/tracing_support.cc
+++ b/content/public/browser/tracing_support.cc
@@ -17,41 +17,38 @@ namespace {
 // other types of tracks.
 static constexpr uint64_t kProcessTrackMagic = 0x12E91CFDD3E3509Bul;
 
-std::optional<base::trace_event::TrackRegistration<perfetto::NamedTrack>>&
+std::optional<base::trace_event::TrackRegistration<perfetto::NamedTrack>> &
 GetWebContentsListTrackRegistrationStorage() {
   static base::NoDestructor<
       std::optional<base::trace_event::TrackRegistration<perfetto::NamedTrack>>>
       page_group_track;
   return *page_group_track;
 }
-
-}  // namespace
+} // namespace
 
 perfetto::Track GetChildProcessTracingTrack(ChildProcessId process_id) {
   return perfetto::Track::Global(base::HashInts(
       kProcessTrackMagic, std::hash<ChildProcessId>()(process_id)));
 }
 
-perfetto::NamedTrack CreateTracingTrackUnderChildProcess(
-    ChildProcessId process_id,
-    perfetto::StaticString name,
-    uint64_t id) {
+perfetto::NamedTrack
+CreateTracingTrackUnderChildProcess(ChildProcessId process_id,
+                                    perfetto::StaticString name, uint64_t id) {
   return perfetto::NamedTrack(name, id,
                               GetChildProcessTracingTrack(process_id));
 }
 
-perfetto::StateTrack GetLocalFrameTracingTrack(
-    const blink::LocalFrameToken& frame_token,
-    bool is_main_frame,
-    ChildProcessId process_id) {
+perfetto::StateTrack
+GetLocalFrameTracingTrack(const blink::LocalFrameToken &frame_token,
+                          bool is_main_frame, ChildProcessId process_id) {
   return blink::GetLocalFrameTracingTrack(
       frame_token, is_main_frame, GetChildProcessTracingTrack(process_id));
 }
 
-perfetto::StateTrack GetWebContentsTracingTrack(
-    const WebContents::UniqueToken& web_contents_token,
-    perfetto::StaticString name) {
-  auto& page_group_track = GetWebContentsListTrackRegistrationStorage();
+perfetto::StateTrack
+GetWebContentsTracingTrack(const WebContents::UniqueToken &web_contents_token,
+                           perfetto::StaticString name) {
+  auto &page_group_track = GetWebContentsListTrackRegistrationStorage();
   if (!page_group_track.has_value()) {
     page_group_track.emplace(
         perfetto::NamedTrack::Global("WebContentsList", 0));
@@ -71,4 +68,4 @@ void ResetWebContentsListTrackRegistrationForTesting() {
   GetWebContentsListTrackRegistrationStorage().reset();
 }
 
-}  // namespace content
+} // namespace content
