diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 70e8a05e3adff08e66bcd2d9b1f4a89922f442ef..d472f7058816e5090a624e14614d961c99af812b 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -21106,6 +21106,9 @@ Please help our engineers fix this problem. Tell us what happened right before y
     <message name="IDS_AHOI_DEVELOPER_ASSET_SCOPE_TAB" desc="Developer asset scope limited to the current tab." meaning="AhoiBrowser developer toolkit">
       Current tab
     </message>
+    <message name="IDS_AHOI_COMMAND_BAR_CURRENT_TAB" desc="Accessible description for the currently active browser tab in AhoiBrowser's command bar." meaning="AhoiBrowser command bar">
+      Current tab
+    </message>
     <message name="IDS_AHOI_DEVELOPER_ASSET_SCOPE_ORIGIN" desc="Developer asset scope limited to the current exact origin." meaning="AhoiBrowser developer toolkit">
       Exact origin
     </message>
diff --git a/chrome/app/resources/generated_resources_de.xtb b/chrome/app/resources/generated_resources_de.xtb
index 823e03fb24409ca970645a1ef8cb5740386079d3..fd4357899aa1a233f19138b3edeaec5b5b51e285 100644
--- a/chrome/app/resources/generated_resources_de.xtb
+++ b/chrome/app/resources/generated_resources_de.xtb
@@ -11168,6 +11168,7 @@ Informationen dazu, welche Daten von Google erhoben werden und warum, findest du
 <translation id="1534074917150785690">Verschlüsselten Sync für dieses Asset erlauben</translation>
 <translation id="956437958128566241">URL-Geltungsbereich</translation>
 <translation id="5720325548335527619">Aktueller Tab</translation>
+<translation id="3062223442052030743">Aktueller Tab</translation>
 <translation id="7592279450846348181">Exakter Ursprung</translation>
 <translation id="6364878297740434630">Domain und Subdomains</translation>
 <translation id="2056065112898830396">Pfadpräfix</translation>
diff --git a/chrome/app/resources/generated_resources_en-GB.xtb b/chrome/app/resources/generated_resources_en-GB.xtb
index 64b905b866cfdb921a4b7931fef52ae4199170b2..d1b664f09d3c65e090bd29fb812e8295d11376d4 100644
--- a/chrome/app/resources/generated_resources_en-GB.xtb
+++ b/chrome/app/resources/generated_resources_en-GB.xtb
@@ -11020,6 +11020,7 @@ This can include information about installed software, files, your browser, the
 <translation id="1534074917150785690">Allow encrypted sync for this asset</translation>
 <translation id="956437958128566241">URL scope</translation>
 <translation id="5720325548335527619">Current tab</translation>
+<translation id="3062223442052030743">Current tab</translation>
 <translation id="7592279450846348181">Exact origin</translation>
 <translation id="6364878297740434630">Domain and subdomains</translation>
 <translation id="2056065112898830396">Path prefix</translation>
