diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index fed0944a5e2a63eb09469ad52b1f118e58349fff..31339e56292338eb393c84444c4fcead5e0c8c24 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -21272,6 +21272,9 @@ Please help our engineers fix this problem. Tell us what happened right before y
     <message name="IDS_AHOI_DEVELOPER_ASSET_SCOPE_TAB" desc="Developer asset scope limited to the current tab." meaning="AhoiBrowser developer toolkit">
       Current tab
     </message>
+    <message name="IDS_AHOI_COMMAND_BAR_CURRENT_TAB" desc="Accessible description for the currently active browser tab in AhoiBrowser's command bar." meaning="AhoiBrowser command bar">
+      Current tab
+    </message>
     <message name="IDS_AHOI_DEVELOPER_ASSET_SCOPE_ORIGIN" desc="Developer asset scope limited to the current exact origin." meaning="AhoiBrowser developer toolkit">
       Exact origin
     </message>
diff --git a/chrome/app/resources/generated_resources_de.xtb b/chrome/app/resources/generated_resources_de.xtb
index ab99943a93f35eb4330c857adbc88f6b09c10999..2724107a811eacc8fbc733b2afa00a731499cadb 100644
--- a/chrome/app/resources/generated_resources_de.xtb
+++ b/chrome/app/resources/generated_resources_de.xtb
@@ -11233,6 +11233,7 @@ Informationen dazu, welche Daten von Google erhoben werden und warum, findest du
 <translation id="1534074917150785690">Verschlüsselten Sync für dieses Asset erlauben</translation>
 <translation id="956437958128566241">URL-Geltungsbereich</translation>
 <translation id="5720325548335527619">Aktueller Tab</translation>
+<translation id="3062223442052030743">Aktueller Tab</translation>
 <translation id="7592279450846348181">Exakter Ursprung</translation>
 <translation id="6364878297740434630">Domain und Subdomains</translation>
 <translation id="2056065112898830396">Pfadpräfix</translation>
diff --git a/chrome/app/resources/generated_resources_en-GB.xtb b/chrome/app/resources/generated_resources_en-GB.xtb
index ed31b7d28fb7dcbd1ba43ea824da6c5c14d47ac5..a278deb6a106fa63a03a6dde33b22426fd3cbe16 100644
--- a/chrome/app/resources/generated_resources_en-GB.xtb
+++ b/chrome/app/resources/generated_resources_en-GB.xtb
@@ -11085,6 +11085,7 @@ This can include information about installed software, files, your browser, the
 <translation id="1534074917150785690">Allow encrypted sync for this asset</translation>
 <translation id="956437958128566241">URL scope</translation>
 <translation id="5720325548335527619">Current tab</translation>
+<translation id="3062223442052030743">Current tab</translation>
 <translation id="7592279450846348181">Exact origin</translation>
 <translation id="6364878297740434630">Domain and subdomains</translation>
 <translation id="2056065112898830396">Path prefix</translation>
