diff --git a/components/embedder_support/user_agent_utils.cc b/components/embedder_support/user_agent_utils.cc
index 608bea5cb386438757146a531bbc0293b3eb9775..c83523ffc037f05803d80fa34527869dd8127f04 100644
--- a/components/embedder_support/user_agent_utils.cc
+++ b/components/embedder_support/user_agent_utils.cc
@@ -163,9 +163,10 @@ const blink::UserAgentBrandList GetUserAgentBrandList(
   bool parse_result = base::StringToInt(major_version, &major_version_number);
   DCHECK(parse_result);
   std::optional<std::string> brand;
-#if !BUILDFLAG(CHROMIUM_BRANDING)
-  brand = version_info::GetProductName();
-#endif
+  const std::string product_brand(version_info::GetProductName());
+  if (product_brand != "Chromium") {
+    brand = product_brand;
+  }
 
   std::string brand_version =
       output_version_type == blink::UserAgentBrandVersionType::kFullVersion
