diff --git a/components/sessions/core/command_storage_manager.cc b/components/sessions/core/command_storage_manager.cc
index cbc0bfc968a0fba8923e8ab27c3b9adfca788217..158b7ce52230f709da678f8881bdeedf64cceca7 100644
--- a/components/sessions/core/command_storage_manager.cc
+++ b/components/sessions/core/command_storage_manager.cc
@@ -473,8 +473,11 @@ void CommandStorageManager::SaveAndReadCurrentSessionCommands(
           base::BindOnce(
               &CommandStorageBackend::FlushAndReadCurrentSessionCommands,
               backend_.get()),
-          base::BindOnce(&CommandStorageManager::OnCurrentSessionReceiptRead,
-                         weak_factory_.GetWeakPtr(), std::move(callback)));
+          base::BindOnce(
+              &OnBackendReadFinished, /*saved_result=*/nullptr,
+              base::BindOnce(
+                  &CommandStorageManager::OnCurrentSessionReceiptRead,
+                  weak_factory_.GetWeakPtr(), std::move(callback))));
       return;
     case EncryptSessionStorageStage::kWriteBothReadOnlyClear:
     case EncryptSessionStorageStage::kWriteBothReadPreferEncrypted:
@@ -484,8 +487,11 @@ void CommandStorageManager::SaveAndReadCurrentSessionCommands(
               &FlushAndReadDualBackends, backend_, encrypted_backend_,
               encrypt_stage ==
                   EncryptSessionStorageStage::kWriteBothReadPreferEncrypted),
-          base::BindOnce(&CommandStorageManager::OnCurrentSessionReceiptRead,
-                         weak_factory_.GetWeakPtr(), std::move(callback)));
+          base::BindOnce(
+              &OnBackendReadFinished, /*saved_result=*/nullptr,
+              base::BindOnce(
+                  &CommandStorageManager::OnCurrentSessionReceiptRead,
+                  weak_factory_.GetWeakPtr(), std::move(callback))));
       return;
     case EncryptSessionStorageStage::kWriteEncryptedReadPreferEncrypted:
       backend_task_runner_->PostTaskAndReplyWithResult(
@@ -493,8 +499,11 @@ void CommandStorageManager::SaveAndReadCurrentSessionCommands(
           base::BindOnce(
               &CommandStorageBackend::FlushAndReadCurrentSessionCommands,
               encrypted_backend_.get()),
-          base::BindOnce(&CommandStorageManager::OnCurrentSessionReceiptRead,
-                         weak_factory_.GetWeakPtr(), std::move(callback)));
+          base::BindOnce(
+              &OnBackendReadFinished, /*saved_result=*/nullptr,
+              base::BindOnce(
+                  &CommandStorageManager::OnCurrentSessionReceiptRead,
+                  weak_factory_.GetWeakPtr(), std::move(callback))));
       return;
   }
 }
