diff --git a/chrome/app/chrome_main_delegate.cc b/chrome/app/chrome_main_delegate.cc
index b2a9613f10ac9ee29cf2d1553cc65906a298bdb8..debcf3c5a84d22305e8127def359dcc944cf6896 100644
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -1127,7 +1127,9 @@ std::optional<int> ChromeMainDelegate::BasicStartupComplete() {
   base::CommandLine& mutable_command_line =
       *base::CommandLine::ForCurrentProcess();
   if (!mutable_command_line.HasSwitch(switches::kProcessType)) {
-    ahoi::startup::ApplyEarlyStartupPolicy(mutable_command_line);
+    if (!ahoi::startup::ApplyEarlyStartupPolicy(mutable_command_line)) {
+      return 1;
+    }
   }
   const base::CommandLine& command_line = mutable_command_line;
 #endif  // !defined(BUILDING_CHROME_RENDERER)
