diff --git a/chrome/app/chrome_main_delegate.cc b/chrome/app/chrome_main_delegate.cc
index 10b22c602c45760e6f17dd6eb644660c6a902ed9..bf937fcc58f38a3c4e788e09d7b45d3f679da775 100644
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -1120,7 +1120,9 @@ std::optional<int> ChromeMainDelegate::BasicStartupComplete() {
   base::CommandLine& mutable_command_line =
       *base::CommandLine::ForCurrentProcess();
   if (!mutable_command_line.HasSwitch(switches::kProcessType)) {
-    ahoi::startup::ApplyEarlyStartupPolicy(mutable_command_line);
+    if (!ahoi::startup::ApplyEarlyStartupPolicy(mutable_command_line)) {
+      return 1;
+    }
   }
   const base::CommandLine& command_line = mutable_command_line;
 #endif  // !defined(BUILDING_CHROME_RENDERER)
