diff --git a/components/privacy_sandbox_strings.grd b/components/privacy_sandbox_strings.grd
index c5c4f59..a63b6f7 100644
--- a/components/privacy_sandbox_strings.grd
+++ b/components/privacy_sandbox_strings.grd
@@ -305,7 +305,7 @@
         </then>
         <else>
           <message name="IDS_SETTINGS_COOKIES_BLOCK_THIRD_PARTY_SETTINGS_ALLOW_BULLET_THREE" desc="This text is for a bullet that provides detail about the first setting option. It describes the key new detail that now applies to this setting option." formatter_data="android_java">
-            When you're in Incognito mode, Chromium blocks sites from using third-party cookies
+            When you're in Incognito mode, AhoiBrowser blocks sites from using third-party cookies
           </message>
         </else>
       </if>
@@ -343,7 +343,7 @@
         </then>
         <else>
           <message name="IDS_PRIVACY_GUIDE_COOKIE_SETTINGS_ALLOW_THINGS_TO_CONSIDER_BULLET_TWO" desc="This text is for a bullet that provides key considerations about the first setting option. It describes the key new detail that now applies to this setting option.">
-            When you're in Incognito mode, Chromium blocks sites from using third-party cookies
+            When you're in Incognito mode, AhoiBrowser blocks sites from using third-party cookies
           </message>
         </else>
       </if>
diff --git a/components/strings/privacy_sandbox_strings_de.xtb b/components/strings/privacy_sandbox_strings_de.xtb
index 8102bc7..14a7511 100644
--- a/components/strings/privacy_sandbox_strings_de.xtb
+++ b/components/strings/privacy_sandbox_strings_de.xtb
@@ -25,6 +25,6 @@
 <translation id="7538480403395139206">Weitere Informationen zur Option „Drittanbieter-Cookies zulassen“</translation>
 <translation id="8200078544056087897">Websitefunktionen, die auf Drittanbieter-Cookies angewiesen sind, sollten wie erwartet funktionieren</translation>
 <translation id="839994149685752920">Websites können Drittanbieter-Cookies verwenden, um Inhalte und Werbung zu personalisieren</translation>
-<translation id="9039924186462989565">Wenn du im Inkognitomodus bist, verhindert Chromium, dass Websites Drittanbieter-Cookies verwenden</translation>
+<translation id="8512038333608422976">Wenn du im Inkognitomodus bist, verhindert AhoiBrowser, dass Websites Drittanbieter-Cookies verwenden</translation>
 <translation id="9162335340010958530">Websites können keine Drittanbieter-Cookies verwenden, um Inhalte und Werbung zu personalisieren oder Informationen zu deinen Aktivitäten auf anderen Websites zu erhalten – es sei denn, du erlaubst ähnlichen Websites den Zugriff darauf</translation>
 </translationbundle>
\ No newline at end of file
diff --git a/extensions/strings/extensions_chromium_strings.grdp b/extensions/strings/extensions_chromium_strings.grdp
index 49d980c..3d40f3a 100644
--- a/extensions/strings/extensions_chromium_strings.grdp
+++ b/extensions/strings/extensions_chromium_strings.grdp
@@ -2,26 +2,26 @@
 <!-- Extensions-specific Chromium strings (included from extensions_strings.grd). -->
 <grit-part>
   <message name="IDS_EXTENSION_EXTERNAL_INSTALL_PROMPT_TITLE_APP" desc="Titlebar of the prompt window for an externally installed app">
-    Another program on your computer added an app that may change the way Chromium works.
+    Another program on your computer added an app that may change the way AhoiBrowser works.
 
 <ph name="EXTENSION_NAME">$1<ex>Babylon Toolbar</ex></ph>
   </message>
   <message name="IDS_EXTENSION_EXTERNAL_INSTALL_PROMPT_TITLE_EXTENSION" desc="Titlebar of the prompt window for an externally installed extension">
-    Another program on your computer added an extension that may change the way Chromium works.
+    Another program on your computer added an extension that may change the way AhoiBrowser works.
 
 <ph name="EXTENSION_NAME">$1<ex>Babylon Toolbar</ex></ph>
   </message>
   <message name="IDS_EXTENSION_EXTERNAL_INSTALL_PROMPT_TITLE_THEME" desc="Titlebar of the prompt window for an externally installed theme">
-    Another program on your computer added a theme that may change the way Chromium works.
+    Another program on your computer added a theme that may change the way AhoiBrowser works.
 
 <ph name="EXTENSION_NAME">$1<ex>Babylon Toolbar</ex></ph>
   </message>
   <message name="IDS_EXTENSION_EXTERNAL_INITIAL_INSTALL_PROMPT_TITLE_EXTENSION" desc="">
-    <ph name="PARTNER_NAME">$1<ex>Google</ex></ph> has added an extension that may change the way Chromium works.
+    <ph name="PARTNER_NAME">$1<ex>Google</ex></ph> has added an extension that may change the way AhoiBrowser works.
 
 <ph name="EXTENSION_NAME">$2<ex>Babylon Toolbar</ex></ph>
   </message>
   <message name="IDS_EXTENSION_EXTERNAL_INSTALL_PROMPT_ABORT_BUTTON" desc="Button on the external install prompt to remove an extension installed by a third party.">
-    Remove from Chromium
+    Remove from AhoiBrowser
   </message>
 </grit-part>
diff --git a/extensions/strings/extensions_strings_de.xtb b/extensions/strings/extensions_strings_de.xtb
index 869245f..be141ea 100644
--- a/extensions/strings/extensions_strings_de.xtb
+++ b/extensions/strings/extensions_strings_de.xtb
@@ -8,7 +8,7 @@
 <translation id="1076698951459398590">Design aktivieren</translation>
 <translation id="1135328998467923690">Paket ist ungültig: "<ph name="ERROR_CODE" />".</translation>
 <translation id="1140871961407805696">Rückkamera</translation>
-<translation id="1171831736360285680"><ph name="PARTNER_NAME" /> hat eine Erweiterung hinzugefügt, die sich möglicherweise auf die Funktionsweise von Chromium auswirkt.
+<translation id="4069788644744279357"><ph name="PARTNER_NAME" /> hat eine Erweiterung hinzugefügt, die sich möglicherweise auf die Funktionsweise von AhoiBrowser auswirkt.
 
 <ph name="EXTENSION_NAME" /></translation>
 <translation id="1196338895211115272">Privater Schlüssel konnte nicht exportiert werden.</translation>
@@ -50,7 +50,7 @@
 <translation id="3144135466825225871">CRX-Datei konnte nicht ersetzt werden. Überprüfe, ob die Datei verwendet wird.</translation>
 <translation id="3163201441334626963">Unbekanntes Produkt <ph name="PRODUCT_ID" /> von Anbieter <ph name="VENDOR_ID" /></translation>
 <translation id="3174723475300476613">Der Zugriff auf diese Erweiterung wird von deiner Organisation blockiert.</translation>
-<translation id="3183357154200327217">Ein anderes Programm auf deinem Computer hat ein Design hinzugefügt, das sich möglicherweise auf die Funktionsweise von Chromium auswirkt.
+<translation id="7447717518726708583">Ein anderes Programm auf deinem Computer hat ein Design hinzugefügt, das sich möglicherweise auf die Funktionsweise von AhoiBrowser auswirkt.
 
 <ph name="EXTENSION_NAME" /></translation>
 <translation id="3242289508736283383">App mit Manifest-Attribut „kiosk_only“ muss im Chrome OS-Kioskmodus installiert werden</translation>
@@ -66,7 +66,7 @@
 <translation id="3927932062596804919">Ablehnen</translation>
 <translation id="3984413272403535372">Fehler bei der Signatur der Erweiterung</translation>
 <translation id="4014432863917027322">"<ph name="EXTENSION_NAME" />" reparieren?</translation>
-<translation id="4061696872429826774">Ein anderes Programm auf deinem Computer hat eine Erweiterung hinzugefügt, die sich möglicherweise auf die Funktionsweise von Chromium auswirkt.
+<translation id="7676690315146880943">Ein anderes Programm auf deinem Computer hat eine Erweiterung hinzugefügt, die sich möglicherweise auf die Funktionsweise von AhoiBrowser auswirkt.
 
 <ph name="EXTENSION_NAME" /></translation>
 <translation id="4094452115117798451">Der Zugriff auf diese Erweiterungskategorie wird von deiner Organisation blockiert.</translation>
@@ -89,7 +89,7 @@
 <translation id="5185386675596372454">Die aktuelle Version von "<ph name="EXTENSION_NAME" />" wurde deaktiviert, weil sie weitere Berechtigungen erfordert.</translation>
 <translation id="5242724311594467048">"<ph name="EXTENSION_NAME" />" aktivieren?</translation>
 <translation id="5356315618422219272">Appview: <ph name="APPVIEW_TAG_NAME" /></translation>
-<translation id="5405650547142096840">Aus Chromium entfernen</translation>
+<translation id="3964831548934448063">Aus AhoiBrowser entfernen</translation>
 <translation id="5456409301717116725">Die Erweiterung enthält die Schlüsseldatei „<ph name="KEY_PATH" />“. Möchtest du den Vorgang wirklich fortsetzen?</translation>
 <translation id="5485102783864353244">App hinzufügen</translation>
 <translation id="5486326529110362464">Es muss ein Eingabewert für den privaten Schlüssel vorhanden sein.</translation>
@@ -99,7 +99,7 @@
 <translation id="5972529113578162692">Der Administrator dieses Computers hat die Installation von <ph name="EXTENSION_NAME" /> vorgegeben. Eine Deinstallation ist daher nicht möglich.</translation>
 <translation id="6027032947578871493">Unbekanntes Produkt <ph name="PRODUCT_ID" /> von <ph name="VENDOR_NAME" /> (Seriennummer <ph name="SERIAL_NUMBER" />)</translation>
 <translation id="6068932090455285721"><ph name="PRODUCT_NAME" /> von Anbieter <ph name="VENDOR_ID" /></translation>
-<translation id="6204922163963157461">Ein anderes Programm auf deinem Computer hat eine App hinzugefügt, die sich möglicherweise auf die Funktionsweise von Chromium auswirkt.
+<translation id="5256962564391942199">Ein anderes Programm auf deinem Computer hat eine App hinzugefügt, die sich möglicherweise auf die Funktionsweise von AhoiBrowser auswirkt.
 
 <ph name="EXTENSION_NAME" /></translation>
 <translation id="6220413761270491930">Fehler beim Laden der Erweiterung</translation>
