diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -1657,6 +1657,7 @@ source_set("browser") {
     "renderer_host/display_feature.h",
     "renderer_host/document_associated_data.cc",
     "renderer_host/document_associated_data.h",
+    "renderer_host/document_loader_factory_refresh.cc",
     "renderer_host/embedded_frame_sink_impl.cc",
     "renderer_host/embedded_frame_sink_impl.h",
     "renderer_host/embedded_frame_sink_provider_impl.cc",
diff --git a/content/public/browser/BUILD.gn b/content/public/browser/BUILD.gn
--- a/content/public/browser/BUILD.gn
+++ b/content/public/browser/BUILD.gn
@@ -158,6 +158,7 @@ source_set("browser_sources") {
     "digital_identity_provider.cc",
     "digital_identity_provider.h",
     "disallow_activation_reason.h",
+    "document_loader_factory_refresh.h",
     "document_picture_in_picture_window_controller.h",
     "document_ref.cc",
     "document_ref.h",
