diff --git a/content/browser/renderer_host/render_frame_host_impl.cc b/content/browser/renderer_host/render_frame_host_impl.cc
--- a/content/browser/renderer_host/render_frame_host_impl.cc
+++ b/content/browser/renderer_host/render_frame_host_impl.cc
@@ -2078,6 +2078,7 @@ class RenderFrameHostImpl::SubresourceLoaderFactoriesConfig {
       RenderFrameHostImpl& frame) {
     SubresourceLoaderFactoriesConfig result;
     result.origin_ = frame.GetLastCommittedOrigin();
+    result.navigation_id_ = frame.GetNavigationId();
     result.isolation_info_ = frame.GetIsolationInfoForSubresources();
     result.client_security_state_ = frame.BuildClientSecurityState();
     if (frame.coep_reporter_) {
@@ -2133,6 +2134,7 @@ class RenderFrameHostImpl::SubresourceLoaderFactoriesConfig {
       NavigationRequest& navigation_request) {
     SubresourceLoaderFactoriesConfig result;
     result.origin_ = navigation_request.GetOriginToCommit().value();
+    result.navigation_id_ = navigation_request.GetNavigationId();
     result.client_security_state_ =
         navigation_request.BuildClientSecurityStateForCommittedDocument();
     result.ukm_source_id_ = ukm::SourceIdObj::FromInt64(
@@ -2275,6 +2277,8 @@ class RenderFrameHostImpl::SubresourceLoaderFactoriesConfig {
 
   const ukm::SourceIdObj& ukm_source_id() const { return ukm_source_id_; }
 
+  int64_t navigation_id() const { return navigation_id_; }
+
   const net::CookieSettingOverrides& cookie_setting_overrides() const {
     return cookie_setting_overrides_;
   }
@@ -2301,6 +2305,7 @@ class RenderFrameHostImpl::SubresourceLoaderFactoriesConfig {
   network::mojom::TrustTokenOperationPolicyVerdict
       trust_token_redemption_policy_;
   ukm::SourceIdObj ukm_source_id_;
+  int64_t navigation_id_ = 0;
   net::CookieSettingOverrides cookie_setting_overrides_;
   base::UnguessableToken network_restrictions_id_;
   bool renderer_accessible_http_cache_write_enabled_ = false;
@@ -3738,7 +3743,8 @@ bool RenderFrameHostImpl::CreateNetworkServiceDefaultFactory(
           subresource_loader_factories_config,
           "RFHI::CreateNetworkServiceDefaultFactory"),
       subresource_loader_factories_config.ukm_source_id(),
-      std::move(default_factory_receiver));
+      std::move(default_factory_receiver),
+      subresource_loader_factories_config.navigation_id());
 }
 
 std::unique_ptr<blink::PendingURLLoaderFactoryBundle>
@@ -8450,7 +8456,8 @@ void RenderFrameHostImpl::UpdateSubresourceLoaderFactories() {
             subresource_loader_factories_config,
             "RFHI::UpdateSubresourceLoaderFactories"),
         subresource_loader_factories_config.ukm_source_id(),
-        default_factory_remote.InitWithNewPipeAndPassReceiver());
+        default_factory_remote.InitWithNewPipeAndPassReceiver(),
+        subresource_loader_factories_config.navigation_id());
   }
 
   std::unique_ptr<blink::PendingURLLoaderFactoryBundle>
@@ -10531,7 +10538,8 @@ RenderFrameHostImpl::CreateCrossOriginPrefetchLoaderFactoryBundle() {
   bypass_redirect_checks = CreateNetworkServiceDefaultFactoryAndObserve(
       std::move(factory_params),
       subresource_loader_factories_config.ukm_source_id(),
-      pending_default_factory.InitWithNewPipeAndPassReceiver());
+      pending_default_factory.InitWithNewPipeAndPassReceiver(),
+      subresource_loader_factories_config.navigation_id());
 
   return std::make_unique<blink::PendingURLLoaderFactoryBundle>(
       std::move(pending_default_factory),
@@ -13278,7 +13286,8 @@ void RenderFrameHostImpl::CommitNavigation(
                   subresource_loader_factories_config,
                   "RFHI::CommitNavigation"),
               subresource_loader_factories_config.ukm_source_id(),
-              pending_default_factory.InitWithNewPipeAndPassReceiver());
+              pending_default_factory.InitWithNewPipeAndPassReceiver(),
+              subresource_loader_factories_config.navigation_id());
       subresource_loader_factories->set_bypass_redirect_checks(
           bypass_redirect_checks);
     }
@@ -13611,7 +13620,8 @@ void RenderFrameHostImpl::FailedNavigation(
               *navigation_request),
           "RFHI::FailedNavigation"),
       ukm::kInvalidSourceIdObj,
-      default_factory_remote.InitWithNewPipeAndPassReceiver());
+      default_factory_remote.InitWithNewPipeAndPassReceiver(),
+      navigation_request->GetNavigationId());
   subresource_loader_factories =
       std::make_unique<blink::PendingURLLoaderFactoryBundle>(
           std::move(default_factory_remote),
@@ -14541,9 +14551,11 @@ bool RenderFrameHostImpl::CreateNetworkServiceDefaultFactoryAndObserve(
     network::mojom::URLLoaderFactoryParamsPtr params,
     ukm::SourceIdObj ukm_source_id,
     mojo::PendingReceiver<network::mojom::URLLoaderFactory>
-        default_factory_receiver) {
+        default_factory_receiver,
+    std::optional<int64_t> navigation_id) {
   bool bypass_redirect_checks = CreateNetworkServiceDefaultFactoryInternal(
-      std::move(params), ukm_source_id, std::move(default_factory_receiver));
+      std::move(params), ukm_source_id, std::move(default_factory_receiver),
+      navigation_id);
 
   // Add a disconnect handler when Network Service is running
   // out-of-process.
@@ -14581,7 +14593,8 @@ bool RenderFrameHostImpl::CreateNetworkServiceDefaultFactoryInternal(
     network::mojom::URLLoaderFactoryParamsPtr params,
     ukm::SourceIdObj ukm_source_id,
     mojo::PendingReceiver<network::mojom::URLLoaderFactory>
-        default_factory_receiver) {
+        default_factory_receiver,
+    std::optional<int64_t> navigation_id) {
   CHECK(params->request_initiator_origin_lock.has_value());
   const url::Origin request_initiator =
       params->request_initiator_origin_lock.value();
@@ -14599,7 +14612,7 @@ bool RenderFrameHostImpl::CreateNetworkServiceDefaultFactoryInternal(
       url_loader_factory::ContentClientParams(
           GetBrowserContext(), this, GetProcess()->GetDeprecatedID(),
           request_initiator, isolation_info, ukm_source_id,
-          &bypass_redirect_checks),
+          &bypass_redirect_checks, navigation_id),
       devtools_instrumentation::WillCreateURLLoaderFactoryParams::ForFrame(
           this));
   return bypass_redirect_checks;
diff --git a/content/browser/renderer_host/render_frame_host_impl.h b/content/browser/renderer_host/render_frame_host_impl.h
--- a/content/browser/renderer_host/render_frame_host_impl.h
+++ b/content/browser/renderer_host/render_frame_host_impl.h
@@ -3672,7 +3672,8 @@ class CONTENT_EXPORT RenderFrameHostImpl
       network::mojom::URLLoaderFactoryParamsPtr params,
       ukm::SourceIdObj ukm_source_id,
       mojo::PendingReceiver<network::mojom::URLLoaderFactory>
-          default_factory_receiver);
+          default_factory_receiver,
+      std::optional<int64_t> navigation_id = std::nullopt);
 
   // Asks an appropriate `NetworkService` to create a new URLLoaderFactory with
   // the given `params` and binds the factory to the `default_factory_receiver`.
@@ -3689,7 +3690,8 @@ class CONTENT_EXPORT RenderFrameHostImpl
       network::mojom::URLLoaderFactoryParamsPtr params,
       ukm::SourceIdObj ukm_source_id,
       mojo::PendingReceiver<network::mojom::URLLoaderFactory>
-          default_factory_receiver);
+          default_factory_receiver,
+      std::optional<int64_t> navigation_id = std::nullopt);
 
   // Returns true if the ExecuteJavaScript() API can be used on this host.
   // The checks do not apply to ExecuteJavaScriptInIsolatedWorld, nor to
